flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java

```diff
@@ -129,6 +130,13 @@ public DataSource createDataSource(Context context) {
         ZoneId serverTimeZone = getServerTimeZone(config);
         StartupOptions startupOptions = getStartupOptions(config);
 
+        // Batch mode only supports StartupMode.SNAPSHOT.
+        Configuration pipelineConfiguration = context.getPipelineConfiguration();
+        if (pipelineConfiguration != null
+                && pipelineConfiguration.contains(PipelineOptions.PIPELINE_BATCH_MODE_ENABLED)) {
+            startupOptions = StartupOptions.snapshot();
+        }
```

Comment:
Just a suggestion, what about throwing an exception explicitly if one enables batch mode with a non-snapshotting source? That would prevent some silent behavior change of `startupOptions` config.

Alternatively, we may add a interface in `DataSourceFactory` like this to make things clearer:

```java
@PublicEvolving
public interface DataSourceFactory extends Factory {

    /** Creates a {@link DataSource} instance. */
    DataSource createDataSource(Context context);

    /** Checking if this {@link DataSource} could be created in batch mode. */
    boolean supportsBatchPipeline(Context context);
}
```

and verifies it during translating pipeline job graph. WDYT?